store/tikv/gcworker/gc_worker.go

```diff
@@ -484,9 +493,35 @@ func (w *GCWorker) calculateNewSafePoint(now time.Time) (*time.Time, error) {
 			zap.String("leaderTick on", w.uuid),
 			zap.Time("last safe point", *lastSafePoint),
 			zap.Time("current safe point", safePoint))
-		return nil, nil
+		return nil, 0, nil
+	}
+	return &safePoint, safePointValue, nil
+}
+
+// setTiDBServiceSafePoint sets the given safePoint as TiDB's service safePoint to PD, and returns the current minimal
+// service safePoint among all services.
+func (w *GCWorker) setTiDBServiceSafePoint(ctx context.Context, safePoint uint64) (uint64, error) {
+	// Sets TTL to 0 to make it permanently valid. TiDB uses empty serviceID.
+	// TODO: Should we use something like "gc_worker" ad the service name here which represents anything that is
```

Comment:
I think it should. There are no compatible problems for it is a new API, use a name make it more friendly to users.